src/transport/CryptoContext.cpp

```diff
@@ -72,17 +72,43 @@ CHIP_ERROR CryptoContext::InitFromSecret(const ByteSpan & secret, const ByteSpan
         infoLen = sizeof(RSEKeysInfo);
     }
 
+#if CHIP_CONFIG_SECURITY_TEST_MODE
+
+    // If enabled, override the generated session key with a known key pair
+    // to allow man-in-the-middle session key recovery for testing purposes.
+
+#define TEST_SECRET_SIZE 32
+    constexpr uint8_t kTestSharedSecret[TEST_SECRET_SIZE] = CHIP_CONFIG_TEST_SHARED_SECRET_VALUE;
+    static_assert(sizeof(CHIP_CONFIG_TEST_SHARED_SECRET_VALUE) == TEST_SECRET_SIZE,
+                  "CHIP_CONFIG_TEST_SHARED_SECRET_VALUE must be 32 bytes");
+    const ByteSpan & testSalt = ByteSpan(nullptr, 0);
+    (void) info;
+    (void) infoLen;
+
+#pragma message                                                                                                                    \
```

Comment:
We shouldn't emit non-fatal non-suppressable diagnostics in the build.

@tcarmelveilleux @turon

Reply:
```
examples/lock-app/cc13x2x7_26x2x7/main/include/CHIPProjectConfig.h:#define CHIP_CONFIG_SECURITY_TEST_MODE 1
examples/persistent-storage/cc13x2x7_26x2x7/include/CHIPProjectConfig.h:#define CHIP_CONFIG_SECURITY_TEST_MODE 1
examples/platform/qpg/project_include/CHIPProjectConfig.h:#define CHIP_CONFIG_SECURITY_TEST_MODE 1
```